src/auth/auth.controller.ts

```diff
@@ -37,4 +39,15 @@ export class AuthController {
   public async signUpWithEmailAndPassword(@Body() createUserDto: CreateUserDto) {
     return this.authService.signUpWithEmailAndPassword(createUserDto);
   }
+
+  @Get("google")
+  @UseGuards(AuthGuard("google"))
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  async googleAuth(@Req() _req: any) {}
+
+  @Get("google/redirect")
+  @UseGuards(AuthGuard("google"))
+  googleAuthRedirect(@Req() req: any) {
+    return this.authService.googleLogin(req);
```

Comment:
We could rename this to something a bit more descriptive

`import { Profile } from "passport-google-oauth20";`

```suggestion
    return this.authService.loginWithGoogleProvider(req.user as Profile);
```

Reply:
Yeah, it does seem more cleaner that way. 
Thanks for the suggestion